File: session06/GrowArrayTest.cc
#include "GrowArray.hh"
#include <vector>
using namespace std;

void test1(int n) {
	GrowArray a;
	for (int i = 0; i < n; i++)
		a.insertEnd(i);
}

void test2(int n) {
	GrowArray a(n);
	for (int i = 0; i < n; i++)
		a.insertEnd(i);
}

void test3(int n) {
	vector<int> a;
	for (int i = 0; i < n; i++)
		a.push_back(i);
}

void test3(int n) {
	vector<int> a;
	a.reserve(n);
	for (int i = 0; i < n; i++)
		a.push_back(i);
}

int main() {
	const int n = 1000000000;
	test1(n);
	//	test2(n);
	//	test3(n);
}

File: old/session06_GrowArray_LinkedList_Stack_Queue/GrowArray/GrowArrayTest.java
import java.util.ArrayList; // used to be Vector

public class GrowArrayTest {
	
	public static void test1(int n) {
		GrowArray a = new GrowArray();
		for (int i = 0; i < n; i++)
			a.insertEnd(i);
	}

	public void test2(int n) {
		GrowArray a = new GrowArray(n);
		for (int i = 0; i < n; i++)
			a.insertEnd(i);
	}

	public static void test3(int n) {
		ArrayList<Integer> a = new ArrayList<>();
		for (int i = 0; i < n; i++)
			a.add(i);
	}

	public static void test4(int n) {
		ArrayList<Integer> a = new ArrayList<>(n);
		for (int i = 0; i < n; i++)
			a.add(i);
	}

int main() {
	const int n = 1000000000;

	// warm up by running the function once
	test1(n);

	long t0 = System.nanoTime();
	test1(n);
	long t1 = System.nanoTime();
	System.out.println((t1-t0)*1e-9);
	
	//	test2(n);
	//	test3(n);
	// test4(n);
}
